docs/sources/developers/plugins/create-a-grafana-plugin/extend-a-plugin/add-log-queries.md

```diff
@@ -0,0 +1,508 @@
+  }
+
+  // This method can be used to show "context" button based on runtime conditions (for example row model data or plugin settings,...)
+  showContextToggle(row?: LogRowModel): boolean {
+    // If you want to always show toggle, you can just return true
+    if (row && row.searchWords && row.searchWords.length > 0) {
+      return true;
+    }
+  }
+}
+```
+
+### Filtering of fields using Log details
+
+> To be implemented trough data source method
+
+Every log line has an expandable part called "Log details" that can be opened by clicking on the line. Within Log details, [Fields]({{< relref "../../../../explore/logs-integration/#fields" >}}) associated with that log entry are displayed. If the data source implements `modifyQuery?(query: TQuery, action: QueryFixAction): TQuery;` API, then filtering functionality is available for each field. For logs, two filtering options are currently available:
+
+- `ADD_FILTER` - to filter for log lines that include selected fields
+- `ADD_FILTER_OUT` - to filter for log lines that do not include selected fields
+
+```ts
+export class ExampleDatasource implements DataSourceWithSupplementaryQueriesSupport<ExampleQuery> {
+  modifyQuery(query: ExampleQuery, action: QueryFixAction): ExampleQuery {
+    let queryText = query.query ?? '';
+    switch (action.type) {
+      case 'ADD_FILTER':
+        if (action.options?.key && action.options?.value) {
+          // This is a mocked implementation. Be sure to adjust this based on your data source logic.
+          queryText = addLabelToQuery(queryText, action.options.key, '=', action.options.value);
+        }
+        break;
+      case 'ADD_FILTER_OUT':
+        {
+          if (action.options?.key && action.options?.value) {
+            // This is a mocked implementation. Be sure to adjust this based on your data source logic.
+            queryText = addLabelToQuery(queryText, action.options.key, '!=', action.options.value);
+          }
+        }
+        break;
+    }
+    return { ...query, query: queryText };
+  }
+}
+```
+
+### Live tailing
+
+> To be implemented trough data source method and enabled in plugin.json
+
+[Live tailing]({{< relref "../../../../explore/logs-integration/#live-tailing" >}}) is a feature that enables real-time log result streaming using Explore. To enable live tailing for your data source, you need to follow these steps:
+
+1. **Enable streaming in plugin.json**: In your data source plugin's plugin.json file, set the `streaming` attribute to `true`. This allows Explore to recognize and enable live tailing controls for your data source.
+
+```json
+{
+  "type": "datasource",
+  "name": "Example",
+  "id": "example",
+  "logs": true,
+  "streaming": true
+}
+```
+
+2. Ensure that your data source's `query` method can handle queries with `liveStreaming` set to true.
+
+```ts
+export class ExampleDatasource implements DataSourceWithSupplementaryQueriesSupport<ExampleQuery> {
```

Comment:
Same as above, `DataSourceWithSupplementaryQueriesSupport<ExampleQuery>` is probably not needed here.

Reply:
Yes, I'll change that, good catch. Copy-paste issue.